### PR TITLE
refactor: merge batching decorators

### PR DESCRIPTION
**Pull request by Kelton8Z**

### CODE REVIEW

File: jina/executors/decorators.py [hunk 1]
```diff
@@ -277,153 +280,60 @@ def arg_wrapper(*args, **kwargs):
...
+                        data[i],
+                        b_size,
+                        split_over_axis,
+                        yield_slice=yield_slice[i],
+                    )
                 )
-                for i in range(0, slice_nargs)
-            ]
 
+            copy_args = copy.copy(args)
```
Comment (JoanFM) [inline jina/executors/decorators.py:303]: is the copy needed? this can become quite expensive

File: jina/executors/decorators.py [hunk 2]
```diff
@@ -277,153 +280,60 @@ def arg_wrapper(*args, **kwargs):
...
+                            mode='r',
+                            shape=original_arg.shape,
+                        )
+                        new_args[i] = new_memmap[slice_idx]
+                        slice_idx = slice_idx[split_over_axis]
+                        if slice_idx.start is None or slice_idx.stop is None:
+                            slice_idx = None
+                        del new_memmap
+
+                    # TODO: figure out what is ordinal_idx_arg
```
Comment (JoanFM) [inline jina/executors/decorators.py:322]: check NmsLibIndexer in jina-hub to understand the feature

File: tests/unit/executors/test_decorators.py [hunk 3]
```diff
@@ -310,12 +333,16 @@ def __init__(self, batch_size):
             self.batch_size = batch_size
             self.ord_idx = []
 
-        @batching(ordinal_idx_arg=2)
+        @batching(
+            ordinal_idx_arg=2, slice_nargs=1, split_over_axis=0
+        )  # f = batching(ordinal_idx_arg=2) f, args[ordinal_idx_arg] = slice_idx(new_args)
         def f(self, data, ord_idx):
+            print(f'ord_idx : {ord_idx}')
```
Comment (JoanFM) [inline tests/unit/executors/test_decorators.py:340]: clean up this test

File: tests/unit/executors/test_decorators.py [hunk 4]
```diff
@@ -412,6 +445,7 @@ def f(self, data0, data1, data2):
     assert result == 1
 
 
+'''
```
Comment (JoanFM) [inline tests/unit/executors/test_decorators.py:448]: remove if not used

**Review by JoanFM — CHANGES_REQUESTED** [review]

**Review by Yongxuanzhang — COMMENTED** [review]
You can first create a draft PR to fix all the tests then open it.
Need to fix corresponding tests to adapt changes

**Review by JoanFM — APPROVED** [review]

**Pull request merged**